inheritance/src/main/java/inheritance/Restaurant.java

```diff
@@ -1,29 +1,37 @@
 package inheritance;
 
+import javax.sound.sampled.ReverbType;
 import java.util.LinkedList;
 
 public class Restaurant {
 
-    LinkedList<Review> restaurantReviews = new LinkedList<>();
     public String restaurantName;
     public float stars;
     public float priceCategory;
 
+    // add all the reviews in the list
+    LinkedList<Review> restaurantReviews = new LinkedList<>();
+
     public Restaurant(String restaurantName, float stars, int priceCategory) {
         this.restaurantName = restaurantName;
         this.stars = stars;
         this.priceCategory = priceCategory;
     }
 
-    public String toString() {
-        return String.format("%s has a rating of %.1f stars out of 5 and it is %.0f price" +
-                "category", this.restaurantName, this.stars, this.priceCategory);
+    public void addReview(Review review) {
+        float totalStars = this.stars;
+        restaurantReviews.add(review);
+        //update average star rating of a restaurant
+        this.stars = (totalStars + restaurantReviews.getLast().stars)/2;
```

Comment:
The math on this isn't correct. You'll need this.stars = totalNumberOfStars/totalNumberOfReviews. The dividing by 2 isn't correct.